matrix/matrix.py

```diff
@@ -0,0 +1,197 @@
+from dataclasses import dataclass
+
+# TODO:
+#   Add documentation.
+
+
+class Matrix(tuple):
```

Comment:
You dont need to inherit from tuple. 
You just an instance of tuple in the class